iib/workers/tasks/build_merge_index_image.py

```diff
@@ -215,8 +217,9 @@ def handle_merge_request(
     dockerfile_name = 'index.Dockerfile'
 
     with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
-        source_fbc = is_image_fbc(source_from_index_resolved)
-        target_fbc = is_image_fbc(target_index_resolved)
+        with set_registry_token(overwrite_target_index_token, source_from_index):
```

Comment:
This is actually the fix it self - once image is pulled it will not fail - but for safety reasons I added set_registry_token when accessing the image.